numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -423,21 +430,27 @@ def __call__(self, *args, global_range=None, local_range=None):
         # TODO: Enable caching of kernels, but do it using Numba's caching
         # machinery
 
-        kernel = SpirvKernel(self.pyfunc, self.kernel_name)
-        kernel.compile(
-            arg_types=argtypes,
-            debug=self.debug_flags,
-            extra_compile_flags=self.compile_flags,
-        )
+        sig = utils.pysignature(self.pyfunc)
+        kernel = self._cache.load_overload(sig, dpex_target.target_context)
+        if kernel is None:
+            kernel = SpirvKernel(self.pyfunc, self.kernel_name)
 
-        # create a sycl::KernelBundle
-        kernel_bundle = dpctl_prog.create_program_from_spirv(
-            exec_queue,
-            kernel.device_driver_ir_module,
-            " ".join(self._create_sycl_kernel_bundle_flags),
-        )
-        #  get the sycl::kernel
-        kernel = kernel_bundle.get_sycl_kernel(kernel.module_name)
+            kernel.compile(
+                arg_types=argtypes,
+                debug=self.debug_flags,
+                extra_compile_flags=self.compile_flags,
+            )
+
+            # create a sycl::KernelBundle
+            kernel_bundle = dpctl_prog.create_program_from_spirv(
+                exec_queue,
+                kernel.device_driver_ir_module,
+                " ".join(self._create_sycl_kernel_bundle_flags),
+            )
+            #  get the sycl::kernel
+            kernel = kernel_bundle.get_sycl_kernel(kernel.module_name)
+
+            self._cache.save_overload(sig, kernel, dpex_target.target_context)
```

Comment:
As discussed, for now let us cache the spir-v module and not the `sycl::kernel_bundle` wrapper